app/services/compute_resource_host_importer.rb

```diff
@@ -0,0 +1,69 @@
+class ComputeResourceHostImporter
+  attr_accessor :host, :compute_resource, :vm
+
+  delegate :logger, :to => :Rails
+
+  def initialize(opts = {})
+    self.compute_resource = opts.fetch(:compute_resource)
+    self.vm = compute_resource.find_vm_by_uuid(opts[:uuid]) if opts[:uuid]
+    self.vm ||= opts.fetch(:vm)
+    self.host = Host::Managed.new(
+      :managed => true,
+      :compute_resource_id => compute_resource.id,
+      :vm => vm
+    )
+    copy_vm_attributes
+    parse_vm_name
+    initialize_interfaces
+  end
+
+  private
+
+  def copy_vm_attributes
+    compute_resource.provided_attributes.each do |foreman_attr, fog_attr|
+      value = vm.public_send(fog_attr)
+      host.send("#{foreman_attr}=", value)
+    end
+  end
+
+  def parse_vm_name
+    name, domain = vm_name.split('.', 2)
+    return unless Net::Validations.validate_hostname(name)
+    domain = Domain.find_by_name(domain)
```

Comment:
since we are importing the vm, should we create the domain if it doesn't exists (just like other fact importers)?